### PR TITLE
CDAP-2965: Update docs to reflect new cdap-client API

### PR DESCRIPTION
**Pull request by jwang47**
https://issues.cask.co/browse/CDAP-2965
http://builds.cask.co/browse/CDAP-DUT2640

### CODE REVIEW

File: cdap-docgen/pom.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      <groupId>com.google.inject</groupId>
+      <artifactId>guice</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.google.code.javaparser</groupId>
+      <artifactId>javaparser</artifactId>
+    </dependency>
+  </dependencies>
+
+</project>
```
Comment (anwar6953) [inline cdap-docgen/pom.xml]: missing newline

File: cdap-docs/reference-manual/source/java-client-api.rst [hunk 2]
```diff
@@ -335,39 +327,20 @@ QueryClient
...
-      header[index] = column.getName() + ": " + column.getType();
-    }
+  ListenableFuture<ExploreExecutionResult> resultFuture = queryClient.execute(Id.Namespace.DEFAULT, "SELECT * FROM dataset_history WHERE customer IN ('Alice','Bob')");
+  ExploreExecutionResult results = resultFuture.get();
+
+  // Fetch schema
+  List<ColumnDesc> schema = results.getResultSchema();
+  String[] header = new String[schema.size()];
+  for (int i = 0; i < header.length; i++) {
+  ColumnDesc column = schema.get(i);
```
Comment (anwar6953) [inline cdap-docs/reference-manual/source/java-client-api.rst]: indentation would have to be preserved in the same way as `nl()`?

Comment (jwang47) [inline cdap-docs/reference-manual/source/java-client-api.rst]: Indentation isn't preserved unfortunately. But yes, we can do begin/endIndent().

File: cdap-docs/reference-manual/source/java-client-api.rst [hunk 3]
```diff
@@ -272,19 +265,19 @@ DatasetClient
...
-  datasetClient.truncate("someDataset");
+  datasetClient.truncate(datasetId);
 
   // Delete a dataset
-  datasetClient.delete("someDataset");
+  datasetClient.delete(datasetId);
 
-
-.. _dataset-module-client:
+.. _datasetmodule-client:
```
Comment (awholegunch) [inline cdap-docs/reference-manual/source/java-client-api.rst]: Why is this changing? ``dataset-module-client`` is correct.

File: cdap-docs/reference-manual/source/java-client-api.rst [hunk 4]
```diff
@@ -296,17 +289,17 @@ DatasetModuleClient
...
-  DatasetModuleMeta datasetModuleMeta = datasetModuleClient.get("someDatasetModule");
+  DatasetModuleMeta datasetModuleMeta = datasetModuleClient.get(datasetModuleId);
 
   // Delete all dataset modules
-  datasetModuleClient.deleteAll();
-
+  datasetModuleClient.deleteAll(Id.Namespace.DEFAULT);
 
-.. _dataset-type-client:
+.. _datasettype-client:
```
Comment (awholegunch) [inline cdap-docs/reference-manual/source/java-client-api.rst]: ibid.

Comment (awholegunch) [inline cdap-docs/reference-manual/source/java-client-api.rst]: If these change, the links to them break.

**Comment by awholegunch** [comment]
I don't quite understand the point of this. Using Java to generate rst seems very heavy-weight. As you see, the formatting is hard to get right. What about using Python? That's what we are doing elsewhere in the build system. Plus, it looks fragile and difficult to maintain.

Another thing to think about is that instead of generating and inserting into the main rst file, you generate a portion of it. That portion is written to a separate file (can be rst, txt, anything) and is then included.
Then you don't have to worry about changes to your file, and potentially the generation can be part of the build. But given how long it takes a JVM to spin up at times...

**Comment by awholegunch** [comment]
Also, as we have a ``cdap-docs`` module; what about using ``cdap-docs-gen``?

**Comment by awholegunch** [comment]
The more I read this...I can see having Java code to build and test...part of the maven build...and then run through a simple Python script that grabs the correct portions into an rst file. Much simpler, easy to integrate, even I could maintain it :)

**Comment by jwang47** [comment]
True, the code that reads the .java and generates the .rst is very hacky here. I think the .java that contains the sample usage is useful, but we should definitely find a better way to generate the .rst.

**Comment by awholegunch** [comment]
Here's a proposal then:
- change the module to ``cdap-docs-gen``, to keep with existing names.
- revise the ``GenerateClientUsageExample.java`` to remove all the hacks added
- perhaps rename it ``ClientUsageExample.java``
- remove all the code generating the rst in Java
- add a test case to check the ``ClientUsageExample.java``

Basically, just get the Java file ready. I will in another PR add a python script that will be called by the build system to create and link up the .rst. Make sense?

**Comment by jwang47** [comment]
Sounds good.

**Pull request merged**